src/metabase/lib/js/metadata.cljs

```diff
@@ -82,13 +83,27 @@
   {:arglists '([object-type])}
   keyword)
 
+(defmulti ^:private rename-key-fn
+  "Returns a function of the keys, either renaming each one or preserving it.
+  If this function returns nil for a given key, the original key is preserved.
+  Use [[excluded-keys]] to drop keys from the input.
+
+  Defaults to nil, which means no renaming is done."
+  identity)
+
+(defmethod rename-key-fn :default [_]
+  nil)
```

Comment:
Wouldn't it be simpler if it defaulted to `identity`? That seems to be the straightforward way to say "no change".

Reply:
Doing that makes the renaming functions harder to write. With this approach they can be maps called like functions, and things that don't appear in the map are preserved.